src/components/rating.ts

```diff
@@ -0,0 +1,62 @@
+import {Component, View, NgFor, NgClass, EventEmitter, OnInit} from 'angular2/angular2';
+
+@Component({
+  selector: 'ngb-rating',
+  inputs: ['max', 'rate', 'readonly'],
+  outputs: ['updateRating: rate', 'onHover', 'onLeave']
+})
+@View({
+  template: `
+    <span tabindex="0" (mouseleave)="reset()" aria-valuemin="0" [attr.aria-valuemax]="max" [attr.aria-valuenow]="rate">
+      <template ng-for #r [ng-for-of]="range" #index="index">
+        <span class="sr-only">({{ index < rate ? '*' : ' ' }})</span>
+        <i class="glyphicon" (mouseenter)="hover(index + 1)" (click)="update(index + 1)" [ng-class]="index < rate ? 'glyphicon-star' : 'glyphicon-star-empty'" [title]="r.title" [attr.aria-valuetext]="r.title"></i>
```

Comment:
I would add the ability to bind to the `tabindex` here since there are no anchor tags.